From 20b4503be31339c72fd716fc03ffee8634ed9d0e Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Wed, 27 Mar 2019 18:45:33 +0800
Subject: [PATCH 495/741] arm64: dts: ls1028a: Add properties for Mali DP500
 node

commit 964cc6d7dfea20fecf8ad30436ea590c254facc1 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The LS1028A has a LCD controller and Displayport interface that
connects to eDP and Displayport connectors on the LS1028A board.

This patch enables the LCD controller driver on the LS1028A.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
Signed-off-by: Wen He <wen.he_1@nxp.com>
Reviewed-by: Liviu Dudau <liviu.dudau@arm.com>
Reviewed-by: Rob Herring <robh@kernel.org>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 38 +++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 1648884b4c3c..680cb6ebf650 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -69,6 +69,27 @@
 		clock-frequency = <100000000>;
 		clock-output-names = "sysclk";
 	};
+ 
+        dpclk: clock-dp {
+                compatible = "fixed-clock";
+                #clock-cells = <0>;
+                clock-frequency = <27000000>;
+                clock-output-names= "dpclk";
+        };
+
+        aclk: clock-axi {
+                compatible = "fixed-clock";
+                #clock-cells = <0>;
+                clock-frequency = <650000000>;
+                clock-output-names= "aclk";
+        };
+
+        pclk: clock-apb {
+                compatible = "fixed-clock";
+                #clock-cells = <0>;
+                clock-frequency = <650000000>;
+                clock-output-names= "pclk";
+        };
 
 	reboot {
 		compatible ="syscon-reboot";
@@ -543,4 +564,21 @@
 			};
 		};
 	};
+
+        display0: malidp@f080000 {
+                compatible = "arm,mali-dp500";
+                reg = <0x0 0xf080000 0x0 0x10000>;
+                interrupts = <0 222 IRQ_TYPE_LEVEL_HIGH>,
+                             <0 223 IRQ_TYPE_LEVEL_HIGH>;
+                interrupt-names = "DE", "SE";
+                clocks = <&dpclk>, <&aclk>, <&aclk>, <&pclk>;
+                clock-names = "pxlclk", "mclk", "aclk", "pclk";
+                arm,malidp-output-port-lines = /bits/ 8 <8 8 8>;
+ 
+                port {
+                        dp0_out: endpoint {
+ 
+                        };
+                };
+        };
 };
-- 
2.17.1

